plugins/manganato/index.js

```diff
@@ -0,0 +1,127 @@
+const SEARCH_URL = "https://manganato.com/search/story/";
+const MAIN_URL = "https://readmanganato.com/";
+const ALT_URL = "https://manganato.com/";
+
+function searchManga(query) {
+    const res = mango.get(SEARCH_URL + query.replace(" ", "_"));
+    if (res.status_code !== 200)
+        mango.raise('Failed to search for manga. Status ' + res.status_code);
+    const manga = mango.css(res.body, "div.search-story-item");
+
+    return JSON.stringify(manga.map(function (m) {
+        const ch = {
+            id: mango.attribute(mango.css(m, "a.item-title")[0], "href").split("/").pop(),
+            title: mango.text(mango.css(m, "a.item-title")[0]),
+            cover_url: mango.attribute(mango.css(m, "img.img-loading")[0], "src")
+        };
+        return ch;
+    }));
+}
+
+function getManga(id) {
+    var res = mango.get(MAIN_URL + id);
+    if (res.status_code !== 200)
+        res = mango.get(ALT_URL + id);
+    if (res.status_code !== 200)
+        mango.raise('Failed to get item from ' + MAIN_URL + ' & ' + ALT_URL + '.');
+    return res;
+}
+
+function listChapters(id) {
+    const res = getManga(id);
+    if (res.status_code !== 200)
+        mango.raise('Failed to get chapters. Status ' + res.status_code);
+    const mangaTitle = mango.text(mango.css(res.body, "div.story-info-right h1")[0]);
+    const manga = mango.css(res.body, "div.panel-story-chapter-list li.a-h");
+
+    return JSON.stringify(manga.map(function (m) {
+        //const chapter = getManga(id + '/' + mango.attribute(mango.css(m, "a.chapter-name")[0], "href").split("/").pop());
+        //const pages = mango.css(chapter.body, "div.container-chapter-reader img");
+
+        const obj = {
+            id: id + '/' + mango.attribute(mango.css(m, "a.chapter-name")[0], "href").split("/").pop(),
+            title: mango.text(mango.css(m, "a.chapter-name")[0]),
+            manga_title: mangaTitle,
+            //pages: pages.length,
+            pages: 0,
```

Comment:
Why are we setting it to `0` instead of using the commented code?

Reply:
Because the number of pages isn't supplied the chapter list page like MangaDex.
To get all the page numbers, we'd have to go through and request each page, causing a lot of overhead for manga with a lot of issues.
Considering its a required field, to speed it up, I set it to 0.